applications/nrf_desktop/src/events/hid_event.c

```diff
@@ -8,36 +8,39 @@
 
 #include "hid_event.h"
 
+#define HID_EVENT_LOG_BUF_LEN 128
+
 
 static int log_hid_report_event(const struct event_header *eh, char *buf,
 				size_t buf_len)
 {
 	const struct hid_report_event *event = cast_hid_report_event(eh);
 	int pos;
+	char log_buf[HID_EVENT_LOG_BUF_LEN];
 
 	__ASSERT_NO_MSG(event->dyndata.size > 0);
 
-	pos = snprintf(buf, buf_len, "Report 0x%x src:%p sub:%p:",
+	pos = snprintf(log_buf, HID_EVENT_LOG_BUF_LEN, "Report 0x%x src:%p sub:%p:",
 		       event->dyndata.data[0],
 		       event->source,
 		       event->subscriber);
-	if ((pos > 0) && (pos < buf_len)) {
+	if ((pos > 0) && (pos < HID_EVENT_LOG_BUF_LEN)) {
 		for (size_t i = 1; i < event->dyndata.size; i++) {
-			int tmp = snprintf(&buf[pos], buf_len - pos,
+			int tmp = snprintf(&log_buf[pos], HID_EVENT_LOG_BUF_LEN - pos,
 					   " 0x%.2x", event->dyndata.data[i]);
 			if (tmp < 0) {
-				pos = tmp;
-				break;
+				EVENT_MANAGER_LOG(eh, "hid_report_event Event Manager logging function failed");
+				return tmp;
 			}
 
 			pos += tmp;
 
-			if (pos >= buf_len)
+			if (pos >= HID_EVENT_LOG_BUF_LEN)
 				break;
 		}
 	}
-
-	return pos;
```

Comment:
No error handling at all?
I would also add the ~ sign as the last one in string when log is incomplete because of too small buffer (to be consistent with previous implementation that indicated cutting log when buffer was too small). Here it would be even more important, because we have dyndata that can have actually any length in customer's usecases

Reply:
The error log could be printed outside for loop to avoid returning inside for loop body